src/shared/filters/api-exception.filter.ts

```diff
@@ -0,0 +1,310 @@
+import { 
+    Catch, 
+    ArgumentsHost, 
+    BadRequestException, 
+    UnauthorizedException, 
+    ForbiddenException, 
+    NotFoundException,
+    MethodNotAllowedException,
+    NotAcceptableException,
+    RequestTimeoutException,
+    ConflictException,
+    GoneException,
+    ExceptionFilter,
+    PayloadTooLargeException,
+    UnsupportedMediaTypeException,
+    UnprocessableEntityException
+} from '@nestjs/common';
+import { HttpAdapterHost } from '@nestjs/core';
+import { ApiException } from '../api-exception.model';
+
+@Catch(BadRequestException)
+class BadRequestExceptionFilter implements ExceptionFilter {
+    constructor(private readonly httpAdapterHost: HttpAdapterHost) {}
+
+    catch(exception: BadRequestException, host: ArgumentsHost): void{
+        const { httpAdapter } = this.httpAdapterHost;
+  
+        const ctx = host.switchToHttp();
+    
+        const responseBody: ApiException = {
+          statusCode: exception.getStatus(),
+          message: "Bad Request - Please check the request sent.",
```

Comment:
Please remove - Bad Request. Http status code should define that. Message should be generic.

Reply:
done